xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/table/livetabledate.js

```diff
@@ -0,0 +1,93 @@
+    'daterangepicker': ['jquery', 'bootstrap', 'moment']
+  }
+});
+
+require(['jquery', 'moment', 'daterangepicker'],
+  function($, moment) {
+    var dateFormat = moment().toMomentFormatString('$escapetool.javascript($xwiki.getXWikiPreference("dateformat", "yyyy/MM/dd HH:mm"))');
+
+    var bindInputs = function(bind) {
+      $(bind).find('input[data-type="date"]').each(function(i, element) {
+        var input = $(element);
+        var hidden = input.prev('input[type="hidden"]');
+        input.daterangepicker({
+          drops: 'down',
+          opens: 'center',
+          autoUpdateInput: false,
+          timePicker: true,
+          timePicker24Hour: true,
+          ranges: {
+            '$escapetool.javascript($services.localization.render("daterange.today"))':
+              [moment().startOf('day'), moment().endOf('day')],
+            '$escapetool.javascript($services.localization.render("daterange.yesterday"))':
+              [moment().subtract(1, 'days').startOf('day'), moment().subtract(1, 'days').endOf('day')],
+            '$escapetool.javascript($services.localization.render("daterange.lastSevenDays"))':
+              [moment().subtract(6, 'days').startOf('day'), moment().endOf('day')],
+            '$escapetool.javascript($services.localization.render("daterange.lastThirtyDays"))':
+              [moment().subtract(29, 'days').startOf('day'), moment().endOf('day')],
+            '$escapetool.javascript($services.localization.render("daterange.thisMonth"))':
+              [moment().startOf('month'), moment().endOf('month')],
+            '$escapetool.javascript($services.localization.render("daterange.lastMonth"))':
+              [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')]
+          },
+          locale: {
+            format: dateFormat,
+            cancelLabel: '$escapetool.javascript($services.localization.render("daterange.clear"))',
+            applyLabel: '$escapetool.javascript($services.localization.render("daterange.apply"))',
+            customRangeLabel: '$escapetool.javascript($services.localization.render("daterange.customRange"))',
+            fromLabel: '$escapetool.javascript($services.localization.render("daterange.from"))',
+            toLabel: '$escapetool.javascript($services.localization.render("daterange.to"))',
+          }
+        });
+
+        var updateInput = function(element, event, picker) {
+          if (event.type == 'cancel') {
+            input.val('');
+            hidden.val('');
+          } else {
+            if (picker.startDate.isSame(picker.endDate, 'day')) {
+              input.val(picker.startDate.format(dateFormat));
+            } else {
+              input.val(picker.startDate.format(dateFormat) + ' - ' + picker.endDate.format(dateFormat));
+            }
+            hidden.val(picker.startDate.format('x') + '-' + picker.endDate.format('x'));
+          }
+          if ('dispatchEvent' in element) {
```

Comment:
I found http://stackoverflow.com/questions/5351442/jquery-change-not-firing-prototype-event-handlers . So I would trigger the event both ways:

    // Trigger Prototype.js listeners because the live table is written using Prototype.js
    element.fire('change');
    // Trigger jQuery listeners.
    $(element).change();